src/Lib.P2P/Swarm.cs

```diff
@@ -0,0 +1,989 @@
+            swarmCancellation?.Cancel(true);
+
+            log.Debug($"Stopping {LocalPeer}");
+
+            // Stop the listeners.
+            while (listeners.Count > 0) await StopListeningAsync(listeners.Keys.First()).ConfigureAwait(false);
+
+            // Disconnect from remote peers.
+            Manager.Clear();
+            Manager.PeerDisconnected -= OnPeerDisconnected;
+
+            otherPeers.Clear();
+            listeners.Clear();
+            pendingConnections.Clear();
+            pendingRemoteConnections.Clear();
+            BlackList = new MultiAddressBlackList();
+            WhiteList = new MultiAddressWhiteList();
+
+            log.Debug($"Stopped {LocalPeer}");
+        }
+
+        /// <summary>
+        ///   Connect to a peer using the specified <see cref="MultiAddress"/>.
+        /// </summary>
+        /// <param name="address">
+        ///   An ipfs <see cref="MultiAddress"/>, such as
+        ///  <c>/ip4/104.131.131.82/tcp/4001/ipfs/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ</c>.
+        /// </param>
+        /// <param name="cancel">
+        ///   Is used to stop the task.  When cancelled, the <see cref="TaskCanceledException"/> is raised.
+        /// </param>
+        /// <returns>
+        ///   A task that represents the asynchronous operation. The task's result
+        ///   is the <see cref="PeerConnection"/>.
+        /// </returns>
+        /// <remarks>
+        ///   If already connected to the peer and is active on any address, then
+        ///   the existing connection is returned.
+        /// </remarks>
+        public async Task<PeerConnection> ConnectAsync(MultiAddress address,
+            CancellationToken cancel = default(CancellationToken))
+        {
+            var peer = RegisterPeerAddress(address);
+            return await ConnectAsync(peer, cancel).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        ///   Connect to a peer.
+        /// </summary>
+        /// <param name="peer">
+        ///  A peer to connect to.
+        /// </param>
+        /// <param name="cancel">
+        ///   Is used to stop the task.  When cancelled, the <see cref="TaskCanceledException"/> is raised.
+        /// </param>
+        /// <returns>
+        ///   A task that represents the asynchronous operation. The task's result
+        ///   is the <see cref="PeerConnection"/>.
+        /// </returns>
+        /// <remarks>
+        ///   If already connected to the peer and is active on any address, then
+        ///   the existing connection is returned.
+        /// </remarks>
+        public async Task<PeerConnection> ConnectAsync(Peer peer, CancellationToken cancel = default(CancellationToken))
+        {
+            if (!IsRunning) throw new Exception("The swarm is not running.");
+
+            peer = RegisterPeer(peer);
+
+            // If connected and still open, then use the existing connection.
+            if (Manager.TryGet(peer, out var conn)) return conn;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)